File: python/006.py
"""
Project Euler - Problem Solution 005
Copyright (c) Justin McGettigan. All rights reserved.
https://github.com/jwmcgettigan/project-euler-solutions
"""

def sum_of_squares(nums) -> int:
  return sum(map(lambda num: num ** 2, nums))

def square_of_sum(nums) -> int:
  return sum(nums) ** 2

def sum_square_difference(nums) -> int:
  return square_of_sum(nums) - sum_of_squares(nums)

if __name__ == "__main__":
	print(sum_square_difference(range(1, 101)))

File: solutions/006/006.js
/**
 * Project Euler - Problem Solution 006
 * Copyright (c) Justin McGettigan. All rights reserved.
 * https://github.com/jwmcgettigan/project-euler-solutions
 */
const { range } = require('../../helpers/helpers');

const sum = (nums) => nums.reduce((a, b) => a + b);

const sumOfSquares = (nums) => {
  return sum(nums.map(num => num*num));
};

const squareOfSum = (nums) => {
  return Math.pow(sum(nums), 2);
};

const sumSquareDifference = (nums) => {
  return squareOfSum(nums) - sumOfSquares(nums);
};

console.log(sumSquareDifference(range(1, 100)));